File: sdk/rust/examples/order_commitment.rs
use shipping_oracle_sdk::{OracleClient, ShipmentReference};

#[derive(Debug)]
struct OrderContext {
    order_id: String,
}

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let base_url =
        std::env::var("ORACLE_BASE_URL").unwrap_or_else(|_| "http://127.0.0.1:3000".to_string());
    let shipment = ShipmentReference::new("shippo", "SHIPPO_DELIVERED");
    let client = match std::env::var("ORACLE_PUBLIC_KEY") {
        Ok(public_key) => OracleClient::new(&base_url).with_expected_public_key_hex(&public_key)?,
        Err(_) => OracleClient::new(&base_url),
    };

    let commitment = client
        .prepare_for(
            OrderContext {
                order_id: "ord_123".to_string(),
            },
            &shipment,
        )
        .await?;

    let tx_args = commitment.to_cli_args_json()?;
    println!("linked order: {}", commitment.context.order_id);
    println!("shipment status: {}", commitment.attestation.data.status);
    println!("tx3 args:\n{}", tx_args.as_json_string());

    Ok(())
}

File: sdk/typescript/examples/order-commitment.ts
/**
 * Example: link an application order to an on-chain shipment tracking commitment.
 *
 * Mirrors sdk/rust/examples/order_commitment.rs.
 *
 * Usage:
 *   ORACLE_BASE_URL=http://127.0.0.1:3000 \
 *   ORACLE_PUBLIC_KEY=<hex> \
 *   node --loader ts-node/esm examples/order-commitment.ts
 *
 * Or with tsx:
 *   pnpm exec tsx examples/order-commitment.ts
 */

// In-repo example imports from source. In your own project, import from the
// package instead: `import { OracleClient } from "shipping-oracle-sdk";`
import { OracleClient } from "../src/index.js";

interface OrderContext {
  orderId: string;
}

async function main(): Promise<void> {
  const baseUrl =
    process.env["ORACLE_BASE_URL"] ?? "http://127.0.0.1:3000";
  const expectedPublicKeyHex = process.env["ORACLE_PUBLIC_KEY"];

  const client = new OracleClient(baseUrl, { expectedPublicKeyHex });

  const commitment = await client.prepareCommitment<OrderContext>(
    { orderId: "ord_123" },
    "shippo",
    "SHIPPO_DELIVERED"
  );

  const { context, attestation } = commitment;

  console.log("linked order   :", context.orderId);
  console.log("status         :", attestation.data.status);
  console.log("carrier_hash   :", attestation.data.carrier_hash);
  console.log("timestamp      :", attestation.data.timestamp);
  console.log("public_key     :", attestation.public_key);
}

main().catch((err: unknown) => {
  console.error("error:", err);
  process.exit(1);
});
